music21/expressions.py

```diff
@@ -1750,19 +2260,25 @@ def realize(self, srcObj: note.Note, *, inPlace=False):
             useQL = self.quarterLength
             fourthNoteQL = opFrac(turnDuration - (3 * useQL))
 
-        if not self.isInverted:
-            firstTransposeInterval = self.getUpperSize(srcObj.pitch)
-            secondTransposeInterval = self.getLowerSize(srcObj.pitch)
+        if not self._isInverted:
+            firstTransposeInterval = self.getSize(srcObj, 'upper', keySig=currentKeySig)
+            secondTransposeInterval = self.getSize(srcObj, 'lower', keySig=currentKeySig)
         else:
-            firstTransposeInterval = self.getLowerSize(srcObj.pitch)
-            secondTransposeInterval = self.getUpperSize(srcObj.pitch)
+            firstTransposeInterval = self.getSize(srcObj, 'lower', keySig=currentKeySig)
+            secondTransposeInterval = self.getSize(srcObj, 'upper', keySig=currentKeySig)
 
-        turnNotes: list[note.Note] = []
+        # no need to check both intervals, they will both be perfectUnison, or neither will be.
+        isTransposed: bool = firstTransposeInterval != interval.Interval('P1')
```

Comment:
ah! here's (and once above) is the other place to avoid creating an Interval just to check if it's a unison.